tiny_dnn/util/serialization_functions.h

```diff
@@ -242,6 +242,90 @@ struct LoadAndConstruct<tiny_dnn::slice_layer> {
   }
 };
 
+template <>
+struct LoadAndConstruct<tiny_dnn::sigmoid_layer> {
+  template <class Archive>
+  static void load_and_construct(
```

Comment:
@edgarriba @karandesai-96 Yeees, we can always use preprocessor macros to reduce redundant template c++ code and dig deeper into nightmarish unreadable code hell. :rage1: